### PR TITLE
[tune] Move reporter detection to utility function

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by Yard1 — APPROVED** [review]
Looks great!

File: python/ray/tune/progress_reporter.py [hunk 1]
```diff
@@ -836,3 +852,20 @@ def _print_result(self, result: Dict):
         print_result_str = ",".join(
             [f"{k}={v}" for k, v in print_result.items()])
         return print_result_str
+
+
+def detect_reporter(**kwargs):
```
Comment (richardliaw) [inline python/ray/tune/progress_reporter.py:857]: add type hints?

**Review by richardliaw — APPROVED** [review]
looks good! are there any tests relevant to add? otherwise merge at will

**Comment by krfricke** [comment]
I added a short unit test. I don't think we're testing Jupyter end to end anywhere.

**Pull request merged**